neqo-http3/src/connection.rs

```diff
@@ -1040,6 +1125,28 @@ impl Http3Events {
     fn insert(&self, event: Http3Event) {
         self.events.borrow_mut().insert(event);
     }
+
+    fn remove_events_for_stream_id(&self, remove_stream_id: u64) {
+        let updated_events = self
+            .events
+            .replace(BTreeSet::new())
+            .iter()
+            .filter(|evt| match evt {
+                Http3Event::HeaderReady { stream_id }
+                | Http3Event::DataWritable { stream_id }
+                | Http3Event::DataReadable { stream_id }
+                | Http3Event::NewPushStream { stream_id }
+                | Http3Event::Reset { stream_id, .. }
+                | Http3Event::StopSending { stream_id, .. } => *stream_id != remove_stream_id,
+                Http3Event::AuthenticationNeeded
+                | Http3Event::GoawayReceived
+                | Http3Event::StateChange { .. }
+                | Http3Event::RequestsCreatable => true,
+            })
+            .cloned()
+            .collect::<BTreeSet<_>>();
```

Comment:
This seems like a lot of work.  If we use a HashSet, we could use `.retain()`.  BTrees don't respond well to that sort of treatment.

Reply:
Currently we rely on events being in order, actually only HeaderReady and DataReadable. This could be refactored, but I would suggest to open a new issue.

Reply:
opened #195